plugins/extractors/bigquery/bigquery.go

```diff
@@ -257,10 +257,37 @@ func (e *Extractor) buildAsset(ctx context.Context, t *bigquery.Table, md *bigqu
 	tableURN := plugins.BigQueryURN(t.ProjectID, t.DatasetID, t.TableID)
 
 	tableProfile := e.buildTableProfile(tableURN, tableStats)
-
 	var partitionField string
+	partitionData := make(map[string]interface{}, 0)
 	if md.TimePartitioning != nil {
-		partitionField = md.TimePartitioning.Field
+		timePartitionMap := make(map[string]interface{}, 0)
+		if md.TimePartitioning.Field != "" {
+			partitionField = md.TimePartitioning.Field
+		} else {
+			partitionField = "_PARTITIONTIME"
+		}
+		timePartitionMap["partition_by"] = string(md.TimePartitioning.Type)
+		timePartitionMap["partition_expire"] = md.TimePartitioning.Expiration.Seconds()
+		partitionData["time_partition"] = timePartitionMap
+	}
+
+	if md.RangePartitioning != nil {
+		rangePartitionMap := make(map[string]interface{}, 0)
+		partitionField = md.RangePartitioning.Field
+		rangePartitionMap["start"] = md.RangePartitioning.Range.Start
+		rangePartitionMap["end"] = md.RangePartitioning.Range.End
+		rangePartitionMap["interval"] = md.RangePartitioning.Range.Interval
+		partitionData["range_partition"] = rangePartitionMap
+	}
+	partitionData["partition_field"] = partitionField
```

Comment:
Shouldn't the partition field be stored for both `time_partition` and `range_partition`?

Reply:
Since it is common field I kept it just under partition_data outside time_partition and range_parition

Reply:
But `partitionField` could be `md.TimePartitioning.Field` or `md.RangePartitioning.Field`. Is the value for both going to be the same?

Reply:
I have created a variable partitionField which depending on TimePartitioning or RangePartitioning reads from md.TimePartitioning.Field or md.RangePartitioning.Field. Please refer line 264 to 267 and 276.

Reply:
Yes, I have added the comment having read those specific lines. I am explicitly referring to the scenario where both lines 265 and 276 are executed (when neither `md.TimePartitioning` nor `md.RangePartitioning` is nil). In this case, we would overwrite the `partitionField` with `md.RangePartitioning.Field` and save only that. Is this the intended behaviour?

Reply:
Okay, I thought on a table only 1 kind of partitioning would be possible, that is why I did not consider this scenario.

Reply:
I tried creating a table keeping an integer field for range partition and a timestamp field to time partition, BigQuery lets me select only one of them for partitioning.  Hence this would not be an issue.
<img width="1309" alt="Screenshot 2023-02-20 at 1 10 25 PM" src="https://user-images.githubusercontent.com/19200212/220043052-f1595957-c495-42f4-a96c-a712369246b1.png">

Reply:
Alright, that works. Can we make this explicit in the code by using an if-else to show that there will be one of the 2?

Reply:
Sure!

Reply:
Done